packages/flutter_tools/lib/src/ios/devices.dart

```diff
@@ -364,6 +355,8 @@ class IOSDevice extends Device {
       '--enable-service-port-fallback',
       '--disable-service-auth-codes',
       '--observatory-port=$assumedObservatoryPort',
+      if (interfaceType == IOSDeviceInterface.network)
+        '--observatory-host=${ipv6 ? '::/0' : '0.0.0.0'}',
```

Comment:
add comment for reader

Reply:
Also see https://github.com/flutter/engine/pull/9485.